frontend/appflowy_tauri/src/appflowy_app/utils/document/slate/markdown.ts

```diff
@@ -3,16 +3,28 @@ import { getBeforeRangeAt } from '$app/utils/document/slate/text';
 import { Editor } from 'slate';
 
 export function canHandleToHeadingBlock(event: React.KeyboardEvent<HTMLDivElement>, editor: Editor): boolean {
+  const flag = getMarkdownFlag(event, editor);
+  if (!flag) return false;
+  const isHeadingMarkdown = /^(#{1,3})$/.test(flag);
+
+  return isHeadingMarkdown;
+}
+
+export function canHandleToCheckboxBlock(event: React.KeyboardEvent<HTMLDivElement>, editor: Editor) {
+  const flag = getMarkdownFlag(event, editor);
+  if (!flag) return false;
+
+  const isCheckboxMarkdown = /^(\[\])$/.test(flag);
```

Comment:
FYI, we support convert both `-[]` and `[]` to uncheckbox, `-[x]` and `[x]` to checkbox.